packages/back-end/src/api/environments/putEnvironment.ts

```diff
@@ -25,15 +25,15 @@ export const putEnvironment = createApiRequestHandler(putEnvironmentValidator)(
     if (
       !req.context.permissions.canCreateOrUpdateEnvironment(updatedEnvironment)
     )
-      throw new Error("You don't have permission to update this environment!");
+      req.context.permissions.throwPermissionError();
 
     const updates: Partial<OrganizationInterface> = {
       settings: {
         ...org.settings,
-        environments: [
-          ...environments.filter((env) => env.id !== id),
-          updatedEnvironment,
-        ],
+        environments: environments.reduce(
+          (ret, env) => [...ret, env.id === id ? updatedEnvironment : env],
+          []
+        ),
```

Comment:
Minor code style, but `map` makes a lot more sense here than `reduce`.  I tend to avoid reduce whenever possible since it's usually much harder to read and follow the logic compared to alternatives.

Reply:
100%